### PR TITLE
(PC-33308) feat(Home): use releaseDate to display date on cine offers

### PR DESCRIPTION
**Pull request by imouandjolobe-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-33308

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [X] Made sure my feature is working on web.
- [X] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [X] Written **unit tests** native (and web when implementation is different) for my feature.
- [X] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [X] I am aware of all the best practices and respected them.

## Screenshots

**delete** _if no UI change_

| Platform         | Mockup/After |
| :--------------- | :--------------: |
| Desktop - Chrome |  |
| Desktop - Chrome | Capture d’écran 2024-11-29 à 09 20 00 |
| Desktop - Chrome | Capture d’écran 2024-11-29 à 10 14 30 |

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a
[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4

## Best Practices

<details>
  <summary>Click to expand</summary>

- Remove non-null assertion operators (just like other type assertions, this doesn’t change the runtime behavior of your code, so it’s important to only use `!` when you know that the value can’t be `null` or `undefined`).

Test specific:

- When you see a local variable that is over-written in every test, mock it.
- Prefer `user` to `fireEvent`.

</details>

### CODE REVIEW

**Review by xlecunff-pass — APPROVED** [review]
Beau ticket de refacto 🎉

File: scripts/noUncheckedIndexedAccess_snapshot.txt [hunk 1]
```diff
@@ -30,7 +30,7 @@
 ./src/features/deeplinks/helpers/getScreenFromDeeplink.ts:24
 ./src/features/home/api/helpers/mapVenuesDataAndModules.ts:13
 ./src/features/home/api/helpers/mapVenuesDataAndModules.ts:15
-./src/features/home/components/modules/OffersModule.tsx:78
+./src/features/home/components/modules/OffersModule.tsx:70
```
Comment (nmajorfrances-pass) [inline scripts/noUncheckedIndexedAccess_snapshot.txt:33]: on peut regarder pour l'enlever ? :)

Comment (imouandjolobe-pass) [reply scripts/noUncheckedIndexedAccess_snapshot.txt:33]: enlever quoi ? 🤔

File: src/shared/hook/useOfferDates.native.test.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        ...defaultMovieScreeningOffer.offer,
+        releaseDate: NOVEMBER_12_2020,
+        dates: [],
+      },
+    })
+
+    expect(result.current).toBe('Dès le 12 novembre 2020')
+  })
+
+  it('should return formatted dates for an offer that is not a movie screening', () => {
```
Comment (nmajorfrances-pass) [inline src/shared/hook/useOfferDates.native.test.ts:51]: je comprend pas le métier derrière ce test

**Review by nmajorfrances-pass — APPROVED** [review]

File: scripts/noUncheckedIndexedAccess_snapshot.txt (hunk 1 above)
Comment (nmajorfrances-pass) [reply scripts/noUncheckedIndexedAccess_snapshot.txt:33]: régler le uncheckedIndex pour pouvoir enlever ce eslintdisable :)

Comment (imouandjolobe-pass) [reply scripts/noUncheckedIndexedAccess_snapshot.txt:33]: J'ai testé, il suffirait d'ajouter un useMemo ! Je vais faire un tour sur l'app pour être sûre que le comportement reste le même

Comment (imouandjolobe-pass) [reply scripts/noUncheckedIndexedAccess_snapshot.txt:33]: Merciii

Comment (imouandjolobe-pass) [reply scripts/noUncheckedIndexedAccess_snapshot.txt:33]: ah en fait ce n'est pas le eslint disable qui est en cause mais le @ts-expect-error et celui-ci va demander de repenser l'archi et la transmission de props, c'est un peu trop pour ce ticket qui a déjà bcp de change et de refacto

**Pull request merged**